docs/high/dataset.rst

```diff
@@ -261,6 +261,34 @@ safe to use with very large target selections.  It is supported for the above
    more information.
 
 
+.. warning::
+    Dataset item getter returns new numpy.ndarray. Some might find the following behavior surprising:
+
+   >>> a = 1
+   >>> b = np.zeros(shape=(2,2))
+   >>> f = h5py.File(file_path, 'w') as f
+   >>> dset = f.create_dataset("test",(2,2))
+   >>> dset[0][1] = a
+   >>> print(dset[0][1])
+   >>> 0
+
+   dset[0] returns an in-memory array. This mean that the set item operation (``[1]``) will be operate on in-memory array, not the on-disk dataset
+   This is equivalent to:
+
+   >>> new_array = dset[1]
```

Comment:
You switch from using coordinates `0, 1` to `1, 1` at this point.

I think it would be clearest to use two different coordinates, and a third number for the value you're setting.